commands/polls/vote.js

```diff
@@ -0,0 +1,107 @@
+                {
+                    key: 'desc',
+                    prompt: '(Optional) Do you have more details?',
+                    type: 'string',
+                    default: ' ',
+                    validate: desc => {
+                        if (desc.length < 201 && desc.length > 11) return true;
+                        return 'Polling questions must be between 10 and 200 characters in length.';
+                    }
+                },
+                {
+                    key: 'time',
+                    prompt: '(Optional) How long should the vote last in minutes?',
+                    type: 'integer',
+                    default: 0,
+                    validate: time => {
+                        if (time >= 0 && time <= 60) return true;
+                        return 'Polling time must be between 0 and 60.';
+                    } 
+                }
+            ]
+        });
+    }
+    
+    run(msg, { question, desc, time }) {
+        var emojiList = ['👍','👎','🤷'];
+        var embed = new RichEmbed()
+            .setTitle(question)
+            .setDescription(desc)
+            .setAuthor(msg.author.username, msg.author.displayAvatarURL)
+            .setColor(0xD53C55) // Green: 0x00AE86
+            .setTimestamp();
+            
+        if (time) {
+            embed.setFooter(`The vote has started and will last ${time} minute(s)`)
+        } else {
+            embed.setFooter(`The vote has started and has no end time`)
+        }
+            
+        //msg.delete(); // Remove the user's command message
+        
+        msg.channel.send({embed}) // Use a 2d array?
+            .then(async function (message) {
+                var reactionArray = [];
+                reactionArray[0] = await message.react(emojiList[0]);
+                reactionArray[1] = await message.react(emojiList[1]);
+                reactionArray[2] = await message.react(emojiList[2]);
+                
+                if (time) {
+                    setTimeout(() => {
+                        // Re-fetch the message and get reaction counts
+                        message.channel.fetchMessage(message.id)
+                            .then(async function (message) {
+                                var reactionCountsArray = [];                               
+                                for (var i = 0; i < reactionArray.length; i++) {
+                                    reactionCountsArray[i] = message.reactions.get(emojiList[i]).count-1;
+                                }
+                                
+                                // Find winner(s)
+                                var max = -Infinity, indexMax = [];
+                                for(var i = 0; i < reactionCountsArray.length; ++i)
+                                    if(reactionCountsArray[i] > max) max = reactionCountsArray[i], indexMax = [i];
+                                    else if(reactionCountsArray[i] === max) indexMax.push(i);
```

Comment:
This is really not a good way to structure code.

Reply:
fixed